### PR TITLE
added latest code max_sum rectangle

### PR DESCRIPTION
**Pull request by shraddhavp**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

added the code for maximum sum rectangle

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#937

### CODE REVIEW

**Comment by shraddhavp** [comment]
Please check the latest changes done as per the  changes requested @iamrajiv  and @Kajol-Kumari   
let me know if any changes

File: Python/cp/max_sum.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    rectangle = [[2,  1, -3, -4,  5],
+                 [0,  6,  3,  4,  1],
+                 [2, -2, -1,  4, -5],
+                 [-3,  3,  1,  0,  3]]
+
+    result = max_sub_sub_rectangle(rectangle)
+    assert 18 == result.maxSum
+    print (result)
+    #Time complexity: O(n^3) or O(COL^2*ROW)
+    #Space Complexity: O(ROW)
```
Comment (iamrajiv) [inline Python/cp/max_sum.py]: remove trailing whitespace at the end

File: Python/cp/max_sum.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+if __name__ == '__main__':
+    rectangle = [[2,  1, -3, -4,  5],
+                 [0,  6,  3,  4,  1],
+                 [2, -2, -1,  4, -5],
+                 [-3,  3,  1,  0,  3]]
+
+    result = max_sub_sub_rectangle(rectangle)
+    assert 18 == result.maxSum
+    print (result)
+    #Time complexity: O(n^3) or O(COL^2*ROW)
```
Comment (iamrajiv) [inline Python/cp/max_sum.py]: update this to `# Time complexity: O(n^3) or O(COL^2*ROW)`

File: Python/cp/max_sum.py (hunk 1 above)
Comment (iamrajiv) [inline Python/cp/max_sum.py]: update this to `# Space Complexity: O(ROW) `

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@shraddhavp Do not make new PR for every changes you can make changes to same PR like  #1072 and #1079 and this PR is same.

**Comment by shraddhavp** [comment]
done @iamrajiv

**Review by iamrajiv — APPROVED** [review]

File: Python/cp/max_sum.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            for i in range(rows):
+                temp[i] += rectangle[i][right]
+
+            res = kadanes(temp)
+            if result.maxSum > result.maxSum:
+                result = Result(res.maxSum, left, right, res.start, res.end)
+
+    return result
+
+if __name__ == '__main__':
```
Comment (vatsalkesarwani12) [inline Python/cp/max_sum.py:82]: dynamic input

File: Python/cp/max_sum.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    rectangle = [[2,  1, -3, -4,  5],
+                 [0,  6,  3,  4,  1],
+                 [2, -2, -1,  4, -5],
+                 [-3,  3,  1,  0,  3]]
+
+    result = max_sub_sub_rectangle(rectangle)
+    assert 18 == result.maxSum
+    print (result)
```
Comment (vatsalkesarwani12) [inline Python/cp/max_sum.py]: add sample output in comment

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]

File: Python/cp/max_sum.py (hunk 3 above)
Comment (shraddhavp) [reply Python/cp/max_sum.py:82]: @plazzy99  i did try it with dynamic input but it was taking a lot of time so I had to hard-code it

Comment (vatsalkesarwani12) [reply Python/cp/max_sum.py:82]: that is not an issue 
go with dynamic input

**Comment by shraddhavp** [comment]
@plazzy99  done the changes as per told ,please check

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

**Review by Kajol-Kumari — APPROVED** [review]
Looks good to me :)

**Pull request merged**